rustls/src/server/hs.rs

```diff
@@ -351,6 +351,30 @@ impl ExpectClientHello {
             cx.common.protocol,
         );
 
+        // And supported kx groups
+        suites::reduce_given_kx_groups(
+            &mut suitable_suites,
+            client_hello.namedgroups_extension(),
+            &self
+                .config
+                .provider
+                .supported_kx_group_names()
+                .collect::<Vec<_>>(),
+        );
+
+        if suitable_suites.is_empty() {
+            return Err(cx.common.send_fatal_alert(
+                AlertDescription::HandshakeFailure,
+                PeerIncompatible::NoKxGroupsInCommon,
```

Comment:
If `suitable_suites` goes empty, we don't really know if the lack of common key exchange groups was at fault, right? It might be due to signature algorithm, version or protocol?

Reply:
I think you're right. Could use `NoCipherSuitesInCommon` instead?

Reply:
You are absolutely right. I think it's still worth it to return the more accurate `NoKxGroupsInCommon`. So for now I fixed it by checking that `suitable_suites` was not empty before. Let me know if it's OK or if I should revert this change.